doc/man/man3/NewFilePage_4.3
.TH "NewFilePage_4" 3 "Fri Jun 12 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
NewFilePage_4 \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <NewFilePage_4\&.h>\fP
.PP
Inherits QWidget\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBNewFilePage_4\fP (QWidget *parent=0)"
.br
.ti -1c
.RI "void \fBinitCore\fP ()"
.br
.ti -1c
.RI "void \fBsetFileNameStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBgetFileNameStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetFileExtStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBgetFileExtStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetProjectStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBgetProjectStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetLocStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBgetLocStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetFolderStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBgetFolderStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetCreatedFileStrPtr\fP ()"
.br
.ti -1c
.RI "QString * \fBgetCreatedFileStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBtriggerMutators\fP ()"
.br
.ti -1c
.RI "QString * \fBtoString\fP ()"
.br
.ti -1c
.RI "\fB~NewFilePage_4\fP ()"
.br
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBhandleFolderPbPtrSlot\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "QLineEdit * \fBfileNameLePtr\fP"
.br
.ti -1c
.RI "QStringList * \fBfileExtsStrLstPtr\fP"
.br
.ti -1c
.RI "QComboBox * \fBfileExtCbPtr\fP"
.br
.ti -1c
.RI "QCheckBox * \fBdefaultFileExtChbPtr\fP"
.br
.ti -1c
.RI "QLineEdit * \fBprojectLePtr\fP"
.br
.ti -1c
.RI "QComboBox * \fBlocCbPtr\fP"
.br
.ti -1c
.RI "QLineEdit * \fBfolderLePtr\fP"
.br
.ti -1c
.RI "QFileDialog * \fBfolderDialogPtr\fP"
.br
.ti -1c
.RI "QPushButton * \fBfolderPbPtr\fP"
.br
.ti -1c
.RI "QLineEdit * \fBcreatedFileLePtr\fP"
.br
.ti -1c
.RI "QGridLayout * \fBfolderLayoutPtr\fP"
.br
.ti -1c
.RI "QFormLayout * \fBformLayoutPtr\fP"
.br
.ti -1c
.RI "QGridLayout * \fBouterLayoutPtr\fP"
.br
.ti -1c
.RI "QString * \fBfileNameStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBfileExtStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBprojectStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBlocStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBfolderStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBcreatedFileStrPtr\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 27 of file NewFilePage_4\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "NewFilePage_4::NewFilePage_4 (QWidget *parent = \fC0\fP)"

.PP
Definition at line 3 of file NewFilePage_4\&.cpp\&.
.SS "NewFilePage_4::~NewFilePage_4 ()"

.PP
Definition at line 203 of file NewFilePage_4\&.cpp\&.
.SH "Member Function Documentation"
.PP 
.SS "QString * NewFilePage_4::getCreatedFileStrPtr ()"

.PP
Definition at line 169 of file NewFilePage_4\&.cpp\&.
.SS "QString * NewFilePage_4::getFileExtStrPtr ()"

.PP
Definition at line 92 of file NewFilePage_4\&.cpp\&.
.SS "QString * NewFilePage_4::getFileNameStrPtr ()"

.PP
Definition at line 80 of file NewFilePage_4\&.cpp\&.
.SS "QString * NewFilePage_4::getFolderStrPtr ()"

.PP
Definition at line 150 of file NewFilePage_4\&.cpp\&.
.SS "QString * NewFilePage_4::getLocStrPtr ()"

.PP
Definition at line 131 of file NewFilePage_4\&.cpp\&.
.SS "QString * NewFilePage_4::getProjectStrPtr ()"

.PP
Definition at line 111 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::handleFolderPbPtrSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 44 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::initCore ()"

.PP
Definition at line 56 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::setCreatedFileStrPtr ()"

.PP
Definition at line 156 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::setFileExtStrPtr ()"

.PP
Definition at line 86 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::setFileNameStrPtr ()"

.PP
Definition at line 67 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::setFolderStrPtr ()"

.PP
Definition at line 137 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::setLocStrPtr ()"

.PP
Definition at line 117 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::setProjectStrPtr ()"

.PP
Definition at line 98 of file NewFilePage_4\&.cpp\&.
.SS "QString * NewFilePage_4::toString ()"

.PP
Definition at line 186 of file NewFilePage_4\&.cpp\&.
.SS "void NewFilePage_4::triggerMutators ()"

.PP
Definition at line 175 of file NewFilePage_4\&.cpp\&.
.SH "Member Data Documentation"
.PP 
.SS "QLineEdit* NewFilePage_4::createdFileLePtr\fC [private]\fP"

.PP
Definition at line 41 of file NewFilePage_4\&.h\&.
.SS "QString* NewFilePage_4::createdFileStrPtr\fC [private]\fP"

.PP
Definition at line 52 of file NewFilePage_4\&.h\&.
.SS "QCheckBox* NewFilePage_4::defaultFileExtChbPtr\fC [private]\fP"

.PP
Definition at line 35 of file NewFilePage_4\&.h\&.
.SS "QComboBox* NewFilePage_4::fileExtCbPtr\fC [private]\fP"

.PP
Definition at line 34 of file NewFilePage_4\&.h\&.
.SS "QStringList* NewFilePage_4::fileExtsStrLstPtr\fC [private]\fP"

.PP
Definition at line 33 of file NewFilePage_4\&.h\&.
.SS "QString* NewFilePage_4::fileExtStrPtr\fC [private]\fP"

.PP
Definition at line 48 of file NewFilePage_4\&.h\&.
.SS "QLineEdit* NewFilePage_4::fileNameLePtr\fC [private]\fP"

.PP
Definition at line 32 of file NewFilePage_4\&.h\&.
.SS "QString* NewFilePage_4::fileNameStrPtr\fC [private]\fP"

.PP
Definition at line 47 of file NewFilePage_4\&.h\&.
.SS "QFileDialog* NewFilePage_4::folderDialogPtr\fC [private]\fP"

.PP
Definition at line 39 of file NewFilePage_4\&.h\&.
.SS "QGridLayout* NewFilePage_4::folderLayoutPtr\fC [private]\fP"

.PP
Definition at line 43 of file NewFilePage_4\&.h\&.
.SS "QLineEdit* NewFilePage_4::folderLePtr\fC [private]\fP"

.PP
Definition at line 38 of file NewFilePage_4\&.h\&.
.SS "QPushButton* NewFilePage_4::folderPbPtr\fC [private]\fP"

.PP
Definition at line 40 of file NewFilePage_4\&.h\&.
.SS "QString* NewFilePage_4::folderStrPtr\fC [private]\fP"

.PP
Definition at line 51 of file NewFilePage_4\&.h\&.
.SS "QFormLayout* NewFilePage_4::formLayoutPtr\fC [private]\fP"

.PP
Definition at line 44 of file NewFilePage_4\&.h\&.
.SS "QComboBox* NewFilePage_4::locCbPtr\fC [private]\fP"

.PP
Definition at line 37 of file NewFilePage_4\&.h\&.
.SS "QString* NewFilePage_4::locStrPtr\fC [private]\fP"

.PP
Definition at line 50 of file NewFilePage_4\&.h\&.
.SS "QGridLayout* NewFilePage_4::outerLayoutPtr\fC [private]\fP"

.PP
Definition at line 45 of file NewFilePage_4\&.h\&.
.SS "QLineEdit* NewFilePage_4::projectLePtr\fC [private]\fP"

.PP
Definition at line 36 of file NewFilePage_4\&.h\&.
.SS "QString* NewFilePage_4::projectStrPtr\fC [private]\fP"

.PP
Definition at line 49 of file NewFilePage_4\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for RIDE from the source code\&.


src/NewFilePage_4.cpp
#include "NewFilePage_4.h"

NewFilePage_4::NewFilePage_4(QWidget* parent) : QWidget(parent)
{
    fileNameLePtr = new QLineEdit();
    fileExtsStrLstPtr = new QStringList();
    fileExtCbPtr = new QComboBox();
    defaultFileExtChbPtr = new QCheckBox();
    projectLePtr = new QLineEdit();
    locCbPtr = new QComboBox();
    folderLePtr = new QLineEdit();
    folderDialogPtr = new QFileDialog();
    folderPbPtr = new QPushButton("Browse...");
    createdFileLePtr = new QLineEdit();
    
    fileNameStrPtr = new QString("NewFile");
    projectStrPtr = new QString("ProjectName");
    locStrPtr = new QString("/home/james/NetBeansProjects/ride/");
    folderStrPtr = new QString("/home/james/NetBeansProjects/ride/NewFile.cpp");
    createdFileStrPtr = new QString("");
    
    folderLayoutPtr = new QGridLayout();
    folderLayoutPtr->addWidget(folderLePtr, 0, 0);
    folderLayoutPtr->addWidget(folderPbPtr, 0, 1);
    
    formLayoutPtr = new QFormLayout();
    formLayoutPtr->addRow(tr("File &Name:"), fileNameLePtr);
    formLayoutPtr->addRow(tr("&Extension:"), fileExtCbPtr);
    formLayoutPtr->addRow(tr(""), defaultFileExtChbPtr);
    formLayoutPtr->addRow(tr("&Project:"), projectLePtr);
    formLayoutPtr->addRow(tr("Loca&tion"), locCbPtr);
    formLayoutPtr->addRow(tr("Folder"), folderLayoutPtr);
    formLayoutPtr->addRow(tr("&Created File:"), createdFileLePtr);
    
    initCore();
    
    outerLayoutPtr = new QGridLayout();
    outerLayoutPtr->addLayout(formLayoutPtr, 0, 0);
    
    this->setLayout(outerLayoutPtr);
}


void NewFilePage_4::handleFolderPbPtrSlot()
{
    QString dirName = folderDialogPtr->getExistingDirectory(this, tr("&Open Directory"),
            "/home",
            QFileDialog::ShowDirsOnly
            | QFileDialog::DontResolveSymlinks);
    
    createdFileLePtr->setText(dirName + "/" + fileNameLePtr->text() + fileExtCbPtr->currentText() );
    locStrPtr = new QString(dirName);
}


void NewFilePage_4::initCore()
{
    fileExtsStrLstPtr->push_back(".h");
    fileExtsStrLstPtr->push_back(".hpp");
    fileExtsStrLstPtr->push_back(".c");
    fileExtsStrLstPtr->push_back(".cpp");
    fileExtCbPtr->addItems(*fileExtsStrLstPtr);
    connect(folderPbPtr, SIGNAL(released()), this, SLOT(handleFolderPbPtrSlot()));
}


void NewFilePage_4::setFileNameStrPtr()
{
    if(fileNameLePtr->text() != "")
    {
        fileNameStrPtr = new QString(fileNameLePtr->text() );
    }
    else
    {
        cerr << "Invalid input at NewFilePage_4::setFileNameStrPtr()" << endl;
    }
}


QString* NewFilePage_4::getFileNameStrPtr()
{
    return fileNameStrPtr;
}


void NewFilePage_4::setFileExtStrPtr()
{
    fileExtStrPtr = new QString(fileExtCbPtr->currentText() );
}


QString* NewFilePage_4::getFileExtStrPtr()
{
    return fileExtStrPtr;
}


void NewFilePage_4::setProjectStrPtr()
{
    if(projectLePtr->text() != "")
    {
        projectStrPtr = new QString(projectLePtr->text() );
    }
    else
    {
        cerr << "Invalid input at NewFilePage_4::setProjectStrPtr()" << endl;
    }
}


QString* NewFilePage_4::getProjectStrPtr()
{
    return projectStrPtr;
}


void NewFilePage_4::setLocStrPtr()
{
    if(locCbPtr->currentText() != "")
    {
        cout << "LOC stub at NewFilPage_4::setLocStrPtr()" << endl;
        //this->locStrPtr = new QString(locCbPtr->currentText() );
    }
    else
    {
        cerr << "Invalid input at NewFilePage_4::setLocStrPtr()" << endl;
    }
}


QString* NewFilePage_4::getLocStrPtr()
{
    return locStrPtr;
}


void NewFilePage_4::setFolderStrPtr()
{
    if(folderLePtr->text() != "")
    {
        folderStrPtr = new QString(folderLePtr->text() );
    }
    else
    {
        cerr << "Invalid input at NewFilePage_4::setFolderStrPtr()" << endl;
    }
}


QString* NewFilePage_4::getFolderStrPtr()
{
    return folderStrPtr;
}


void NewFilePage_4::setCreatedFileStrPtr()
{
    if(createdFileLePtr->text() != "")
    {
        createdFileStrPtr = new QString(createdFileLePtr->text() );
    }
    else
    {
        cerr << "Invalid input at NewFilePage_4::setCreatedFileStrPtr()" << endl;
    }
}


QString* NewFilePage_4::getCreatedFileStrPtr()
{
    return createdFileStrPtr;
}


void NewFilePage_4::triggerMutators()
{
    setFileNameStrPtr();
    setFileExtStrPtr();
    setProjectStrPtr();
    setLocStrPtr();
    setFolderStrPtr();
    setCreatedFileStrPtr();
}


QString* NewFilePage_4::toString()
{
    QString* tmp = new QString("File Name: ");
    tmp->append(getFileNameStrPtr() );
    tmp->append("\nProject Name: ");
    tmp->append(getProjectStrPtr() );
    tmp->append("\nLocation: ");
    tmp->append(getLocStrPtr() );
    tmp->append("\nFolder: ");
    tmp->append(getFolderStrPtr() );
    tmp->append("\nCreated File: ");
    tmp->append(getCreatedFileStrPtr() );
    
    return tmp;
}


NewFilePage_4::~NewFilePage_4()
{
    ;
}